src/main/java/com/solace/samples/jcsmp/features/TopicToQueueMapping2.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2021-2022 Solace Corporation. All rights reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.solace.samples.jcsmp.features;
+
+import java.util.concurrent.CountDownLatch;
+
+import com.solacesystems.jcsmp.BytesXMLMessage;
+import com.solacesystems.jcsmp.CapabilityType;
+import com.solacesystems.jcsmp.Consumer;
+import com.solacesystems.jcsmp.ConsumerFlowProperties;
+import com.solacesystems.jcsmp.DeliveryMode;
+import com.solacesystems.jcsmp.EndpointProperties;
+import com.solacesystems.jcsmp.JCSMPException;
+import com.solacesystems.jcsmp.JCSMPFactory;
+import com.solacesystems.jcsmp.JCSMPProperties;
+import com.solacesystems.jcsmp.JCSMPSession;
+import com.solacesystems.jcsmp.JCSMPStreamingPublishCorrelatingEventHandler;
+import com.solacesystems.jcsmp.Queue;
+import com.solacesystems.jcsmp.TextMessage;
+import com.solacesystems.jcsmp.Topic;
+import com.solacesystems.jcsmp.XMLMessageListener;
+import com.solacesystems.jcsmp.XMLMessageProducer;
+
+/**
+ * NOTE: this sample does not illustrate Solace best practices for topic-to-queue subscription
+ * management.  Ideally, best practice is to admin-configure queues with topic subscriptions
+ * using SEMP or PubSub+ Manager.  See example in folder "semp-rest-api".
+ * However, if you want your messaging API to be able to add/remove topic subscriptions on
+ * queues, this sample shows that.  Note that your queue needs "modify topic" permissions
+ * to work, rather than the default "consume" permissions.
+ */
+public class TopicToQueueMapping2  {

```

Comment:
Why 2 of these now? Do they show different ways of doing it or something? Either remove if extra or change file names to be clear how they are different w/out someone having to dig into the code

Reply:
The topic-to-queue mapping example for R&D was super different than the one I had.  Diff'ing didn't work.  I kept both in until I had more time to look into differences and pick one.
.....
Ok, so the diff is b/c when Solace Cloud put up their own (simplified set of) samples on GH, they updated/modernized the one that was included inside the tarball.  Now: our "patterns" or regular Guaranteed messaging samples all use topic-to-queue mapping, as this is best practice and not a "neat feature to maybe use".  At least the SC one is a bit better.  E.g.:

Tarball: `Topic addedTopic = JCSMPFactory.onlyInstance().createTopic("sample_queue_AddedTopic");`
GH: `Topic tutorialTopic = JCSMPFactory.onlyInstance().createTopic("T/mapped/topic/sample");`

After spending some time, I think the GH one is better/newer, and they both do pretty much the same thing.  And this sample will probably disappear at some point anyhow, get rolled up into the "queue provision" sample (since this sample also provisions a queue) and all our Guaranteed samples should be using/expecting topics on queues.